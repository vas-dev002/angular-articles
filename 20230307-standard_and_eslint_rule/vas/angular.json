{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "vas-app": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "inlineTemplate": true,
          "inlineStyle": true,
          "style": "scss",
          "skipTests": true
        },
        "@schematics/angular:class": {
          "skipTests": true
        },
        "@schematics/angular:directive": {
          "skipTests": true
        },
        "@schematics/angular:guard": {
          "skipTests": true
        },
        "@schematics/angular:interceptor": {
          "skipTests": true
        },
        "@schematics/angular:pipe": {
          "skipTests": true
        },
        "@schematics/angular:resolver": {
          "skipTests": true
        },
        "@schematics/angular:service": {
          "skipTests": true
        },
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "root": "projects/vas-app",
      "sourceRoot": "projects/vas-app/src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/vas-app",
            "index": "projects/vas-app/src/index.html",
            "main": "projects/vas-app/src/main.ts",
            "polyfills": "projects/vas-app/src/polyfills.ts",
            "tsConfig": "projects/vas-app/tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "projects/vas-app/src/favicon.ico",
              "projects/vas-app/src/assets"
            ],
            "styles": [
              "projects/vas-app/src/styles.scss"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "projects/vas-app/src/environments/environment.ts",
                  "with": "projects/vas-app/src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "vas-app:build:production"
            },
            "development": {
              "browserTarget": "vas-app:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "vas-app:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/vas-app/**/*.ts",
              "projects/vas-app/**/*.html"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "vas-app",
  "cli": {
    "defaultCollection": "@angular-eslint/schematics"
  }
}
