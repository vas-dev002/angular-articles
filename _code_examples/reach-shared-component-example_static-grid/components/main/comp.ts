import {
	Component, OnDestroy, ChangeDetectionStrategy, Input, HostListener, EventEmitter, Output, Injector
} from "@angular/core";
import {
	Module,
	GridApi,
	GridOptions,
	ColDef,
	ColumnApi,
	CsvExportParams,
	ColumnEvent,
	RowDragEvent,
	Column,
	ProcessCellForExportParams
} from "@ag-grid-community/core";
import { MenuModule } from "@ag-grid-enterprise/menu";
import { ClientSideRowModelModule } from "@ag-grid-community/client-side-row-model";
import { ServerSideRowModelModule } from "@ag-grid-enterprise/server-side-row-model";
import { ClipboardModule } from "@ag-grid-enterprise/clipboard";
import { CsvExportModule } from "@ag-grid-community/csv-export";
import { RowGroupingModule } from "@ag-grid-enterprise/row-grouping";
import { ColumnsToolPanelModule } from "@ag-grid-enterprise/column-tool-panel";
import {
	getMainDef, frameworkComponents,
	getEditDef, getServiceDef, getSelectionDef,
	getValueByType, isRowSelectable, getDataKey, getDragDef
} from "./config/main";
import {
	IColumnMetaSG,
	IOptionsSG,
	IInternalStateSG,
	IDataRowSG,
	defaultDataKeySG,
	IDragAndDropModel,
	GridMode,
	IOptionsDynamicSG
} from "./entity";
import { merge, intersection } from "lodash";
import { Subject, BehaviorSubject, Observable } from "rxjs";
import { ISelectAllMethods } from "../columns";
import { ColumnControllerSGDirective } from "./directives/column.directive";
import { FilterControllerSGDirective } from "./directives/filter.directive";
import { SortControllerSGDirective } from "./directives/sort.directive";
import { SelectControllerSGDirective } from "./directives/select.directive";
import * as moment from "moment";
import { checkClass } from "../../../../../util/common/element";
import { aggridMenuFix } from "../../../../../util/common/aggridMenuFix";
import { StaticGridApiService } from "./services/static-grid-api.service";
import { getGridSortModel } from "../../../util/fn";
import { tableModalWindowFix } from "../../../../../util/common/style";

@Component({
	selector: "static-grid",
	templateUrl: "./tpl.html",
	changeDetection: ChangeDetectionStrategy.OnPush,
	providers: [StaticGridApiService]
})
export class StaticGridComponent implements OnDestroy {

	// Options
	@Input() public set options(value: IOptionsSG) {
		this._options = value;
		this.setupGrid();
	}
	public get options() {
		return this._options;
	}
	private _options: IOptionsSG;

	@Input() public set dynamicModeOptions(value: IOptionsDynamicSG) {
		this._dynamicModeOptions = value;
		this.setupGrid();
	}
	public get dynamicModeOptions() {
		return this._dynamicModeOptions;
	}
	private _dynamicModeOptions: IOptionsDynamicSG;

	@Input() public set defaultConfig(val: GridOptions) {
		this._defaultConfig = val;
		this.setupGrid();
	}
	public get defaultConfig(): GridOptions {
		return this._defaultConfig;
	}
	private _defaultConfig: GridOptions;

	@Input() public gridMode: GridMode = GridMode.StaticMode;
	@Input() public rowData: IDataRowSG[];
	@Input() public hasData: boolean;
	@Input() public disabled = false;

	public columnController: ColumnControllerSGDirective;
	public filterController: FilterControllerSGDirective;
	public sortController: SortControllerSGDirective;
	public selectController: SelectControllerSGDirective;
	public readonly dateFormat = "ll";
	public readonly dateTimeFormat = "lll";
	public gridStyles: any = {};

	// MetaData
	@Input() public set columnMetaData(value: IColumnMetaSG[]) {
		this._columnMetaData = value;
		this.setupGrid();
	}
	public get columnMetaData() {
		return this._columnMetaData;
	}
	private _columnMetaData: IColumnMetaSG[] = [];


	@Output() visibleColumnsChange = new EventEmitter<ColDef[]>();
	@Output() gridReady = new EventEmitter<StaticGridComponent>();
	@Output() dragAndDropUpdate = new EventEmitter<IDragAndDropModel>();
	@Output() cellUpdate = this.staticGridApiService.cellUpdate;

	public destroy$ = new Subject();
	public columnDefs: ColDef[] = [];
	public gridOptions: GridOptions = {};
	public gridApi: GridApi;
	public gridColumnApi: ColumnApi;

	public agGridModules: Module[] = [
		ClipboardModule, ClientSideRowModelModule,
		CsvExportModule, MenuModule,
		RowGroupingModule, ColumnsToolPanelModule, ServerSideRowModelModule
	];

	public dataKey = defaultDataKeySG;

	// Temporary solution. Used to get data in a directive when referring to the component as a class.
	private readonly _firstDataRendered$: BehaviorSubject<boolean> = new BehaviorSubject(null);
	public firstDataRendered$: Observable<boolean> = this._firstDataRendered$.asObservable();

	constructor(private readonly injector: Injector,
		private readonly staticGridApiService: StaticGridApiService) {}

	private setupGrid() {
		if(!this.options || !this.columnMetaData) {
			return;
		}
		this.configureGridOptions();
		this.configureColDef();
	}

	private configureGridOptions(): void {
		this.gridOptions.frameworkComponents = frameworkComponents;
		if (this.options.enableRowDragAndDrop) {
			const dragDropUpdate = (event: RowDragEvent) => {
				const key = event.node.data[this.dataKey];
				const index = event.node.rowIndex + 1;
				this.dragAndDropUpdate.emit({key: key, index: index});
			};
			this.gridOptions.onRowDragEnd = dragDropUpdate;

			// enable managed drag
			this.gridOptions.rowDragManaged = true;
			this.gridOptions.animateRows = true;
		}
		merge(
			this.gridOptions,
			this.defaultConfig,
			this.options.agGridOptions as GridOptions,
			this.dynamicModeOptions?.agGridOptions as GridOptions
		);
	}

	private getRowCount(excludeDisabledRows = false) {
		if (this.rowData) {
			return (!excludeDisabledRows)
				? this.rowData.length
				: this.rowData.filter((row) =>
					isRowSelectable(row, this.columnMetaData)
				).length;
		} else {
			return 0;
		}
	}

	private getSelectedCount() {
		return this.gridApi ? this.gridApi.getSelectedNodes().length : 0;
	}

	private configureColDef() {
		if (this.columnMetaData && this.columnMetaData.length > 0) {
			this.columnDefs = this.columnMetaData.map(x =>
				merge(
					getMainDef(x, this, this.options.columnControllerOptions, this.injector),
					this.options.editorsByKey
						&& this.options.editorsByKey[x.columnName] ? getEditDef(x, this.options.editorsByKey[x.columnName]) : {}
				)
			);

			if (this.options.serviceColumns) {
				this.options.serviceColumns.forEach((x) => {
					const def = getServiceDef(x);
					const index = (x.index > this.columnDefs.length) ? 0 : x.index;
					this.columnDefs.splice(index, 0, def);
				});
			}

			if (this.options.enableCheckBox) {
				const selectMethods: ISelectAllMethods = {
					getRowsCount: (excludeDisabledRows) => this.getRowCount(excludeDisabledRows),
					getSelectedCount: () => this.getSelectedCount(),
					selectAll: () => this.selectController.selectAllRows(),
					deselectAll: () => this.selectController.deselectAllRows(),
					onSelectionChanged: this.selectController.selectedDataRowsChange
				};

				const showingSelectionOptions = {
					showCheckBoxes: this.selectController.showCheckboxes,
					suppressHeaderCheckBox: this.options.suppressHeaderCheckBox
				};

				this.columnDefs.splice(0, 0,
					getSelectionDef(showingSelectionOptions, selectMethods, this.columnMetaData)
				);
			}

			this.rowDragColDef();
		}
	}

	private rowDragColDef() {
		if (this.options.enableRowDragAndDrop) {
			this.columnDefs.splice(0, 0, getDragDef(this.columnMetaData));
		}
	}

	onColumnVisibilityChanged(event: ColumnEvent) {
		this.columnController.updateGroupColumns(this.gridColumnApi.getRowGroupColumns().map((x) => x.getColId()));
		this.columnController.updateColumnState(this.gridColumnApi.getColumnState());

		const visibleColumns = this.columnController.getVisibleColumns();
		this.visibleColumnsChange.emit(visibleColumns.map(x => x.getColDef()));
	}

	public onReady(params) {
		this.dataKey = getDataKey(this.options, this.columnMetaData);
		this.gridApi = params.api;
		this.gridColumnApi = params.columnApi;
		this.columnController.init(this.gridOptions, this.gridApi, this.gridColumnApi, this.gridMode);
		this.filterController?.init(this.gridOptions, this.gridApi, this.gridColumnApi, this.gridMode);
		this.sortController?.init(this.gridOptions, this.gridApi, this.gridColumnApi, this.gridMode);
		this.selectController?.init(this.gridOptions, this.gridApi, this.gridColumnApi, this.gridMode);
		this.gridOptions.onColumnVisible = (event: ColumnEvent) => this.onColumnVisibilityChanged(event);
		this.restoreGridState({
			groupColumns: this.columnController.selectedGroupColumns,
			columns: this.columnController.columnState,
			filters: this.filterController?.selectedFilters,
			sortModel: this.sortController?.sortModel
		});
		this.updateColumnDefs();
		this.gridReady.next(this);

		tableModalWindowFix();
	}

	public onFirstDataRendered(params) {
		this._firstDataRendered$.next(true);
		setTimeout(() => {
			this.adjustGridSize();
		});
	}

	public adjustGridSize() {
		this.setColumnWidth();
	}

	public getGridState(): IInternalStateSG {
		return {
			groupColumns: this.gridColumnApi.getRowGroupColumns().map((x) => x.getColId()),
			columns: this.gridColumnApi.getColumnState(),
			filters: this.gridApi.getFilterModel(),
			sortModel: getGridSortModel(this.gridColumnApi)
		};
	}

	public restoreGridState(state: IInternalStateSG) {
		this.columnController.updateGroupColumns(state.groupColumns);
		this.columnController.updateColumnState(state.columns);
		this.filterController?.updateFilters(state.filters);
		this.sortController?.updateSortModel(state.sortModel);
	}

	public restoreDefaultColumnsVisibility(gridState: IInternalStateSG) {
		gridState.columns
			.filter(column => this.columnMetaData.some(colMeta => colMeta.columnName === column.colId && colMeta.visible))
			.forEach(column => column.hide = false);
	}

	public updateColumnDefs() {
		const state = this.getGridState();
		const hasVisibleColumns = !!this.columnController.getVisibleColumns().length;
		// if all columns were hidden by user - restore all column visibility on refresh
		if (state.columns && !hasVisibleColumns) {
			this.restoreDefaultColumnsVisibility(state);
		}
		this.restoreGridState(state);
	}

	public get emptyMessage(): string {
		return this.options && this.options.emptyMessage;
	}

	private isExcludeFromExport(colDef: ColDef): boolean {
		return !!(colDef.refData && colDef.refData.excludeFromExport);
	}

	public exportData(): void {
		if (!this.gridApi && !this.gridColumnApi) { return; }
		let fileName = "Dashworks export";
		if (!!this.options.exportFileName) {
			fileName = this.options.exportFileName.replace(/[^0-9a-z_ \-\(\)]/gi, "");
		}
		fileName += ` - ` + `${moment().format("YYYYMMDD")}`;

		let columnKeys = this.gridColumnApi.getAllColumns()
			.filter((column: Column) => column.isVisible() && !this.isExcludeFromExport(column.getColDef()))
			.map((column: Column) => column.getColId());

		columnKeys = intersection(columnKeys, this.columnMetaData.map((columnMeta: IColumnMetaSG) => columnMeta.columnName));
		const params: CsvExportParams = {
			allColumns: false,
			fileName,
			columnKeys,
			processCellCallback: (cellForExportParams: ProcessCellForExportParams): string => {
				return getValueByType(cellForExportParams.column.getColDef(), cellForExportParams.value);
			}
		};
		this.gridApi.exportDataAsCsv(params);
	}

	private setColumnWidth() {
		if (this.gridColumnApi) {
			this.gridColumnApi.autoSizeAllColumns();
		}
	}

	@HostListener("window:resize", ["$event"])
	public onResize() {
		this.adjustGridSize();
	}

	@HostListener("document:click", ["$event"])
	onClick(event) {
		if (checkClass(event.target, "ag-header-cell-menu-button") || checkClass(event.target, "mat-option")) {
			aggridMenuFix();
		}
		if (checkClass(event.target, "mat-checkbox") || checkClass(event.target, "ag-menu-option")) {
			aggridMenuFix();
		}
	}

	public ngOnDestroy() {
		this.destroy$.next();
		this.destroy$.complete();
		if (document.getElementById("content")) {
			document.getElementById("content").style.overflow = "auto";
		}
	}
}
