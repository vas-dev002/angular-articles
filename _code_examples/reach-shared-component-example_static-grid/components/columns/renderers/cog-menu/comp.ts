import { Component } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { RowNode } from "@ag-grid-community/core";
import { ICellCogMenuParams, ICogMenuItem } from "./entity";

@Component({
	selector: "child-cell",
	templateUrl: "tpl.html"
})
export class CogMenuSGComponent implements ICellRendererAngularComp {
	public permissionStatus: boolean;
	public params: ICellCogMenuParams;
	public items: ICogMenuItem[];
	public isOpened = false;
	private _getIndex(): number {
		let index = this.params.rowIndex;
		this.params.api.forEachNodeAfterFilterAndSort((rowNode: RowNode, rowIndex: number) => {
			if (this.params.node.id === rowNode.id) {
				index = rowIndex;
			}
		});
		return index;
	}

	public agInit(params: ICellCogMenuParams): void {
		this.params = params;
		this.items = params.menuItems;
	}

	public refresh(): boolean {
		return false;
	}

	public show(item: ICogMenuItem) {
		return item.hide && item.hide({ params: this.params, rowIndex: this._getIndex() });
	}

	public onClick(item: ICogMenuItem) {
		item.onClick({ params: this.params, rowIndex: this._getIndex() });
	}

	public get isDisabled(): boolean {
		return !!this.params && !!this.params.isDisabled && this.params.isDisabled(this.params.data);
	}
}
