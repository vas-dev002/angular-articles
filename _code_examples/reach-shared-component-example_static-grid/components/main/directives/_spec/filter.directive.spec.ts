import { TestBed, ComponentFixture } from "@angular/core/testing";
import { Component, ViewChild } from "@angular/core";
import { StaticGridComponent } from "../../comp";
import { GridOptions, RowNode, GridApi, ColumnApi, Column } from "@ag-grid-community/core";
import { AgGridModule } from "@ag-grid-community/angular";
import { TranslateService, TranslateModule } from "@ngx-translate/core";
import { FilterControllerSGDirective } from "../filter.directive";
import { ColumnState } from "@ag-grid-community/core/dist/es6/columnController/columnController";
import { GridMode, ISelectedFilters } from "../../..";

@Component({
	template: `
		<static-grid
			filterControllerSG=""
            [selectedFilters]="selectedFilters"
            ></static-grid>`
})
class TestComponent {
	@ViewChild(FilterControllerSGDirective, { static: true })
	public filterController: FilterControllerSGDirective;

	columnApi = {
		setRowGroupColumns(colKeys: (string | Column)[]) { },
		setColumnState(columnState: ColumnState[]) { },
		getAllDisplayedColumns() {
			return [] as Column[];
		},
		getColumnState() {
			return [];
		}
	} as ColumnApi;

	gridApi = {
		getFilterModel() {
			return [];
		},
		setFilterModel(model) { },
		forEachNodeAfterFilter(callback: (rowNode: RowNode, index: number) => void) { },
		onFilterChanged() { }
	} as unknown as GridApi;

	gridOptions = {
		columnApi: this.columnApi,
		columnDefs: []
	} as GridOptions;

	mode = GridMode.StaticMode;
}

describe("Static Grid Filter Controller", () => {
	let component: TestComponent;
	let fixture: ComponentFixture<TestComponent>;
	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [TranslateModule.forRoot(), AgGridModule.withComponents([])],
			providers: [TranslateService],
			declarations: [
				StaticGridComponent,
				TestComponent,
				FilterControllerSGDirective
			]
		});
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(TestComponent);
		component = fixture.componentInstance;
		component.filterController.init(
			component.gridOptions,
			component.gridApi,
			component.columnApi,
			component.mode
		);
		fixture.detectChanges();
	});

	it("should compile", () => {
		expect(component).toBeTruthy();
	});

	it("should return selectedFilters when the values are equal", () => {
		const value: ISelectedFilters = {
			"key": {
				type: "type",
				filter: "filter",
				filterType: "filterType"
			}
		};
		const newGridApi = {
			getFilterModel() {
				return value;
			},
			setFilterModel(model) { },
			forEachNodeAfterFilter(callback: (rowNode: RowNode, index: number) => void) { },
			onFilterChanged() { }
		} as GridApi;

		component.filterController.init(
			component.gridOptions,
			newGridApi,
			component.columnApi,
			component.mode
		);
		component.filterController.selectedFilters = value;
		const selectedFilters = component.filterController.selectedFilters;
		expect(selectedFilters).toMatchSnapshot();
	});

	it("should return selectedFilters", () => {
		component.filterController.selectedFilters = {};
		const selectedFilters = component.filterController.selectedFilters;
		expect(selectedFilters).toMatchSnapshot();
	});

	it("should be called onFilterChanged", () => {
		const onFilterChanged = component.gridOptions.onFilterChanged(null);
		expect(onFilterChanged).toMatchSnapshot();
	});

	it("should return the right numbers of displayed rows", () => {
		const nodes = [
			{ group: false }, { group: false }, { group: true }
		] as RowNode[];
		component.gridApi.forEachNodeAfterFilter = (callback) => {
			nodes.forEach(callback);
		};
		const spyOn = jest.spyOn(component.gridApi, "forEachNodeAfterFilter");
		const res = component.filterController.getDisplayedRowCount();
		expect(spyOn).toHaveBeenCalled();
		expect(res).not.toBe(nodes.length);
		expect(res).toBe(2);
	});

	describe("set custom filter", () => {
		it("should return custom filter", () => {
			component.filterController.customFilterRules = () => true;
			const res = component.filterController.customFilterRules;
			expect(res).toBeDefined();
			expect(res({} as RowNode)).toBe(true);
		});

		it("should set custom rules", () => {
			component.filterController.customFilterRules = () => true;
			expect(component.gridOptions.isExternalFilterPresent).toBeDefined();
			expect(component.gridOptions.isExternalFilterPresent()).toBe(true);
			expect(component.gridOptions.doesExternalFilterPass).toBeDefined();
			expect(component.gridOptions.doesExternalFilterPass({} as RowNode)).toBe(true);
		});

		it("should reapply filters", () => {
			const spyOn = jest.spyOn(component.gridApi, "onFilterChanged");
			component.filterController.customFilterRules = () => true;
			expect(spyOn).toHaveBeenCalledTimes(1);
		});

		it("should reset custom rules", () => {
			component.gridOptions.isExternalFilterPresent = () => true;
			component.gridOptions.doesExternalFilterPass = () => true;
			component.filterController.customFilterRules = null;
			expect(component.gridOptions.isExternalFilterPresent).toBeDefined();
			expect(component.gridOptions.isExternalFilterPresent()).toBe(false);
			expect(component.gridOptions.doesExternalFilterPass).toBe(null);
		});

		it("should not call getDisplayedRowCount when grid mode is not StaticMode", () => {
			const spyOnGetDisplayedRowCount = jest.spyOn(component.filterController, "getDisplayedRowCount");
			component.mode = GridMode.DynamicMode;

			component.filterController.init(
				component.gridOptions,
				component.gridApi,
				component.columnApi,
				component.mode
			);
			component.gridOptions.onFilterChanged(null);

			expect(spyOnGetDisplayedRowCount).toHaveBeenCalledTimes(0);
		});
	});
});
