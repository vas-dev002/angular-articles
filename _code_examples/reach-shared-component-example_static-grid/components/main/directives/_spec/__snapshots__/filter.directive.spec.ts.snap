// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Grid Filter Controller should be called onFilterChanged 1`] = `undefined`;

exports[`Static Grid Filter Controller should return selectedFilters 1`] = `Object {}`;

exports[`Static Grid Filter Controller should return selectedFilters when the values are equal 1`] = `
Object {
  "key": Object {
    "filter": "filter",
    "filterType": "filterType",
    "type": "type",
  },
}
`;
